hspec-discover/src/Config.hs

```diff
@@ -24,6 +26,7 @@ options = [
   , Option [] ["formatter"] (ReqArg (\s c -> c {configFormatter = Just s}) "FORMATTER") ""
   , Option [] ["module-name"] (ReqArg (\s c -> c {configModuleName = Just s}) "NAME") ""
   , Option [] ["no-main"] (NoArg $ \c   -> c {configNoMain = True}) ""
+  , Option "h" ["help"] (NoArg $ \c -> c {configHelp = True}) ""
```

Comment:
Following GNU conventions (e.g. check `cp`), I would only add `--help`, but would skip the `-h` shortcut.